### PR TITLE
week1 homework

### PR DESCRIPTION
**Pull request by majidkhan**
I had created a pull request of HackYourFuture earlier. It seems almost forgot Git working process.

### CODE REVIEW

File: week1/homework.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import random
+
+print(' ## Task 1 ## \n')
+#Write a python script to print your name and age
+
+name = "Majid Khan"
+age = int(2021-1983)
+print('My name is ' + name + "\nI am " + str(age) + " years old \n")
```
Comment (Rassibassi) [inline week1/homework.py]: have a look at python f-strings, they are like javascript string templates

File: week1/homework.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+
+print(' ## Task 2 ## \n')
+# Create a list of your 5 favorite movies and store it in the variable
+favMovies = ['The Shawshank Redemption', 'The Dark Knight ', 'Catch Me If You Can', 'The Godfather', 'For a Few Dollars More' ]
+for i in range(len(favMovies)):
+    print('My all time favourite movies are; \n', favMovies[i])
```
Comment (Rassibassi) [inline week1/homework.py]: Python can make your work with for loops way easier :)
checkout my last comment here, for how to handle for loops in Python

File: week1/homework.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+print('\n ## Task 10 ## \n')
+
+# Accept a string from user and remove the characters which have odd index values of a given string and print them.
+
+userStringInput = input('Pleas enter string: ')
+result = ""
+for i in range(len(userStringInput)):
+    if (i % 2 == 0):
+      result = result + userStringInput[i]
```
Comment (Rassibassi) [inline week1/homework.py]: Python can do this more elegant or for a _Pythonista_, there is a more _pythonic_ way of doing this.

In Python **a string is basically a list of characters**, or in other words, in Python like a list also a string is **iterable**,
and a list can be looped through like this:
```python
sample_list = [100,200,300]
for sample_int in sample_list:
    print(sample_int)
```
Do you see that there is no indexing `sample_list[i]` necessary anymore? You can do the same with a string

Now, you might have also seen that we lost the index number `i` within the loop. Yet, there is also a pythonic way of getting a running number into your for loop, see here:
```python
sample_list = [100,200,300]
for i, sample_int in enumerate(sample_list):
    print(i, sample_int)
```

Can you use my feedback and improve your code?

and also:
```Python
a = a + b
```
is the same as
```Python
a += b
```

File: week1/homework.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+citiesList = ['Copenhagen', 'Aarhus', 'Odense', 'Aalborg', 'Roskilde']
+print('List:                        ', citiesList)
+citiesTuple = tuple(citiesList)
+print('List converted into tuple:   ', citiesTuple)
+
+print('\n ## Task 9 ## \n')
+# Remove duplicated from the list and store the values in the same list:
+a = [10,20,30,20,10,50,60,40,80,50,40]
+print('Original list:               ', a)
+a = list(dict.fromkeys(a))
```
Comment (Rassibassi) [inline week1/homework.py]: This works, but if I would not know the internals of the python `dict` then this is difficult to understand. Meaning the `fromkeys` method is not meant to be used this way. checkout `set` or have a look in the little solutions gist I posted in slack

File: week1/homework.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+
+print('\n ## Task 7 ## \n')
+#Create a tuple with different data types
+
+myTuple= (1.1, 'A', '/', 10, 'y', '+', ['mango', 'apple', 'orange'], {"name" : "Majid", "class" : "JSript"})
+
+# print(str(i) + ' is of type' + str(type(i)))
+for i in myTuple:
+    print(str(i) + '        is of type  ' + str(type(i)))
```
Comment (Rassibassi) [inline week1/homework.py]: Check fstrings again!
and having a lot of empty lines in your output strings is bad practice :)

**Review by Rassibassi — COMMENTED** [review]
Good job, see my inline comments

File: week1/homework.py (hunk 4 above)
Comment (majidkhan) [reply week1/homework.py]: That is pretty simple method. But the problem using set is that the original order of the list is lost. What do you suggest in this case?

Comment (majidkhan) [reply week1/homework.py]: I will prefer the imperative programming example :-)

File: week1/homework.py (hunk 5 above)
Comment (majidkhan) [reply week1/homework.py]: code improved

**Pull request merged**